Sources/NIOHTTP2/HTTP2Frame.swift

```diff
@@ -125,33 +125,86 @@ public struct HTTP2Frame: Sendable {
         /// The payload of a `DATA` frame.
         public struct Data {
             /// The application data carried within the `DATA` frame.
-            public var data: IOData
+            public var data: IOData {
+                @inlinable
+                get {
+                    return self._backing.data
+                }
+                set {
+                    self._copyIfNeeded()
+                    self._backing.data = newValue
+                }
+            }
 
             /// The value of the `END_STREAM` flag on this frame.
-            public var endStream: Bool
-
-            /// The underlying number of padding bytes. If nil, no padding is present.
-            internal private(set) var _paddingBytes: UInt8?
+            public var endStream: Bool {
+                @inlinable
+                get {
+                    return self._backing.endStream
+                }
+                set {
+                    self._copyIfNeeded()
+                    self._backing.endStream = newValue
+                }
+            }
 
             /// The number of padding bytes sent in this frame. If nil, this frame was not padded.
             public var paddingBytes: Int? {
                 get {
-                    return self._paddingBytes.map { Int($0) }
+                    return self._backing.paddingBytes.map { Int($0) }
                 }
                 set {
+                    self._copyIfNeeded()
                     if let newValue = newValue {
                         precondition(newValue >= 0 && newValue <= Int(UInt8.max), "Invalid padding byte length: \(newValue)")
-                        self._paddingBytes = UInt8(newValue)
+                        self._backing.paddingBytes = UInt8(newValue)
                     } else {
-                        self._paddingBytes = nil
+                        self._backing.paddingBytes = nil
                     }
                 }
             }
 
+            @usableFromInline
+            var _backing: _Backing
+
             public init(data: IOData, endStream: Bool = false, paddingBytes: Int? = nil) {
-                self.data = data
-                self.endStream = endStream
-                self.paddingBytes = paddingBytes
+                self._backing = _Backing(data: data, endStream: endStream, paddingBytes: paddingBytes.map { UInt8($0) })
+            }
+
+            @inlinable
+            init(_ backing: _Backing) {
+                self._backing = backing
+            }
+
+            @inlinable
+            mutating func _copyIfNeeded() {
+                if !isKnownUniquelyReferenced(&self._backing) {
+                    self._backing = self._backing.copy()
+                }
+            }
```

Comment:
Does this need to be `@inlinable` if it is not public and never called from an `@inlinable` declaration? If not, we could make it `fileprivate`